api/comments/views.py

```diff
@@ -0,0 +1,140 @@
+    """Replies to a comment.
+
+    By default, a GET will return both deleted and not deleted replies. Comment replies may be
+    filtered by their `deleted` field.
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublicForComments,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_COMMENTS_READ]
+    required_write_scopes = [CoreScopes.NODE_COMMENTS_WRITE]
+
+    serializer_class = CommentSerializer
+
+    def get_queryset(self):
+        return Comment.find(Q('target', 'eq', self.get_comment()))
+
+    # overrides ListCreateAPIView
+    def perform_create(self, serializer):
+        self.check_object_permissions(self.request, self.get_comment())
+        serializer.save()
+
+
+class CommentDetail(generics.RetrieveUpdateAPIView, CommentMixin):
+    """Details about a specific comment."""
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublicForComments,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_COMMENTS_READ]
+    required_write_scopes = [CoreScopes.NODE_COMMENTS_WRITE]
+
+    serializer_class = CommentDetailSerializer
+
+    # overrides RetrieveAPIView
+    def get_object(self):
+        return self.get_comment()
+
+
+class CommentReportsList(generics.ListCreateAPIView, CommentMixin):
+    """List of reports made for a comment."""
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        ContributorOrPublicForComments,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_COMMENTS_READ]
+    required_write_scopes = [CoreScopes.NODE_COMMENTS_WRITE]
+
+    serializer_class = CommentReportsSerializer
+
+    def get_queryset(self):
+        user_id = self.request.user._id
+        comment = self.get_comment()
+        reports = comment.reports
+        serialized_reports = []
+        if user_id in reports:
+            report = CommentReport(user_id, reports[user_id]['category'], reports[user_id]['text'])
+            serialized_reports.append(report)
+        return serialized_reports
+
+
+class CommentReportDetail(generics.RetrieveUpdateDestroyAPIView, CommentMixin):
+    """Details about a specific comment report."""
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        ContributorOrPublicForComments,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_COMMENTS_READ]
+    required_write_scopes = [CoreScopes.NODE_COMMENTS_WRITE]
+
+    serializer_class = CommentReportDetailSerializer
+
+    # overrides RetrieveUpdateDestroyAPIView
+    def get_object(self):
+        comment = self.get_comment()
+        reports = comment.reports
+        user_id = self.request.user._id
+        report_id = self.kwargs['user_id']
+
+        if report_id != user_id:
```

Comment:
I can appreciate consistency, but I'm not quite sold on a URL parameter that only has one relevant value per user. It seems like it would add more effort to using the endpoint, without delivering clear benefits or any real customization.

If it helps, we originally had a similar "owner" parameter for applications URLs, but scrapped it for similar design reasons.